src/peer_channels.rs

```diff
@@ -285,15 +286,22 @@ impl Context {
     }
 
     /// Get or initialize the iroh peer channel.
-    pub async fn get_or_try_init_peer_channel(&self) -> Result<&Iroh> {
+    pub async fn get_or_try_init_peer_channel(&self) -> Result<Arc<Iroh>> {
         if !self.get_config_bool(Config::WebxdcRealtimeEnabled).await? {
             bail!("Attempt to get Iroh when realtime is disabled");
         }
 
-        let ctx = self.clone();
-        self.iroh
-            .get_or_try_init(|| async { ctx.init_peer_channels().await })
-            .await
+        {
+            let lock = self.iroh.read().await;
+            if let Some(ref iroh) = *lock {
+                return Ok(Arc::clone(iroh));
+            }
+        }
+
+        let mut lock = self.iroh.write().await;
+        let iroh = Arc::new(self.init_peer_channels().await?);
```

Comment:
`Option` value should be rechecked so that two parallel calls to this function never init peer channels twice